lib/_stream_writable.js

```diff
@@ -108,6 +108,33 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // count buffered requests
+  this.bufferedRequestCount = 0;
+
+  this.corkedRequestsFree = new CorkedRequest(this);
+  this.corkedRequestsFree.next = new CorkedRequest(this);
+}
+
+function CorkedRequest(state) {
```

Comment:
Do you have any thoughts on moving this to the bottom? It might be better than putting it in between the `WritableState` constructor and prototype definitions, even if `CorkedRequest` is then farther away from where it is first used.